### ISSUE TITLE
Rule Change: don't flag max-params for native methods that take more than 3 arguments

### ISSUE THREAD

**Issue by japamatSynapse** [eslint/eslint#15129]
### What rule do you want to change?

max-params

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
const helloWorldAsArray = ['Hello', 'world', '!'];
const helloWorldAsString = helloWorldAsArray.reduce(
  (finalString, curString, idx, arr) => {
    if (idx !== arr.length - 1) {
      return finalString + String(' ') + curString;
    }
    return finalString + curString;
  }
, '')
```

### What does the rule currently do for this code?

it tells me that "Arrow function has too many parameters (4). Maximum allowed is 3", when the native `Array.prototype` method takes up to 4. Would be nice if this could somehow know if this is a callback for an native method that takes more than 3.

### What will the rule do after it's changed?

it will not warn me when im using native methods that take more than 3 parameters

### Participation

- [ ] I am willing to submit a pull request to implement this change.

**Comment by nzakas** [comment]
Thanks for the suggestion. ESLint doesn’t have the information about which native methods have how many parameters, so we aren’t able to do this easily. You can use a disable comment to disable the warning where you need to.
